atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/SingleBusyCellWorkflows.java

```diff
@@ -0,0 +1,110 @@
+import java.util.stream.Stream;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This workflow attempts to manipulate a single cell in a table, scheduling transactions that perform writes and
+ * deletes, along with transactions that simply read. Notice that unlike other workflows, because the load is
+ * considerably heterogeneous, we do not use a {@link DefaultWorkflow} to schedule transactions.
+ */
+public final class SingleBusyCellWorkflows {
+    private SingleBusyCellWorkflows() {
+        // utility
+    }
+
+    @VisibleForTesting
+    static final WorkloadCell BUSY_CELL = ImmutableWorkloadCell.of(1, 1);
+
+    public static Workflow create(
+            InteractiveTransactionStore store,
+            SingleBusyCellWorkflowConfiguration configuration,
+            ListeningExecutorService readExecutor,
+            ListeningExecutorService writeExecutor) {
+        return () -> {
+            List<ListenableFuture<Optional<WitnessedTransaction>>> reads =
+                    scheduleReadsAndTouches(store, configuration, readExecutor);
+            List<ListenableFuture<Optional<WitnessedTransaction>>> writes =
+                    scheduleWrites(store, configuration, writeExecutor);
+
+            ListenableFuture<List<WitnessedTransaction>> witnessedTransactions = Futures.transform(
+                    Futures.allAsList(
+                            Stream.of(reads, writes).flatMap(Collection::stream).collect(Collectors.toList())),
+                    maybeWitnessedTransactions -> maybeWitnessedTransactions.stream()
+                            .flatMap(Optional::stream)
+                            .collect(Collectors.toList()),
+                    MoreExecutors.directExecutor());
+            ReadOnlyTransactionStore readOnlyTransactionStore = new ReadOnlyTransactionStore(store);
+            return ImmutableWorkflowHistory.builder()
+                    .addAllHistory(WitnessedTransactions.sortAndFilterTransactions(
+                            readOnlyTransactionStore, Futures.getUnchecked(witnessedTransactions)))
+                    .transactionStore(readOnlyTransactionStore)
+                    .build();
+        };
+    }
+
+    @NotNull
+    private static List<ListenableFuture<Optional<WitnessedTransaction>>> scheduleReadsAndTouches(
+            InteractiveTransactionStore store,
+            SingleBusyCellWorkflowConfiguration configuration,
+            ListeningExecutorService readExecutor) {
+        return IntStream.range(0, configuration.iterationCount() / 2)
+                .mapToObj(idx -> readExecutor.submit(() -> store.readWrite(txn -> {
+                    Optional<Integer> value =
+                            txn.read(configuration.tableConfiguration().tableName(), BUSY_CELL);
+                    txn.write(configuration.tableConfiguration().tableName(), BUSY_CELL, value.orElse(-1));
+                })))
+                .collect(Collectors.toList());
+    }
+
+    @NotNull
```

Comment:
Don't we opt for `@Nullable` for things that could be _null_ rather than `@NotNull` to specify that it doesn't have nulls?

Reply:
Yeah I'll remove this. This was auto-added by intellij, I should check my formatting rules